src/foam/core/Enum.js

```diff
@@ -315,6 +315,13 @@ foam.CLASS({
         return of && of.VALUES[0];
       },
     },
+    {
+      name: 'javaValue',
+      expression: function(of) {
+        var val = (of && of.VALUES[0]);
+        return foam.String.isInstance(val) ? val : this.of.id + '.' + val;
```

Comment:
L322 is based on the understanding that `isInstance` checks for the fully qualified name (so if it is fully qualified just returns that, otherwise adds package + class -> id). If that understanding of `isInstance` is incorrect, and it only checks for a valid type, then the conditional values need to be reversed. Please let me know.